### ISSUE TITLE
Investigate bad SpamBug classifications

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#1289]
E.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1615701 was classified as spam with high confidence, but it isn't spam.
The first step is to check the feature importances for this classification, so we can see why the model chose the way it did.

**Comment by gabbyprecious** [comment]
Can I work on this?

**Comment by marco-c** [comment]
Sure, feel free.
@ayush-1506 might be interested to investigate too in order to complete the spambug-related project, but this issue can easily be worked on by multiple people since it's an investigation issue.

**Comment by ayush-1506** [comment]
I tried generating the confidence scores this weekend but the terminal crashes with `Killed: 9`. (Which seems to be oom error). I then tried reducing the size of the dataset, but turns out the reduced dataset had only one label every time I tried, so that couldn't work too. Will look into the alternatives.

**Comment by marco-c** [comment]
Probably best to reduce the features considered by the model (the best bet is using a `min_df` for text_vectorizer like we do in other models, you can try with a few different values to see which one doesn't decrease accuracy).

**Comment by Nikhil1O1** [comment]
This one appears to be open, Is it? @marco-c 
If yes I would like to work.
Could you help me pinpoint it in the code base

**Comment by tarunjarvis5** [comment]
Hi, I am new to the community would like to understand and learn to solve the issue. 
Can someone walk me through it?

**Comment by marco-c** [comment]
This issue is not super-easy, as it requires some investigation and understanding of how the project works.
You can start with this one if you want, but you'll need to play around with the code a bit. You can start by training the spambug model (run the scripts/trainer.py script) and classifying a bug with the spambug model (run the scripts/bug_classifier.py script).
Then look at the source code of those scripts and try to understand how they work.

**Comment by alrho007** [comment]
I'll like to work on this issue. 
Permit me to ask you any question if I get stuck or find anything strange.

**Comment by suhaibmujahid** [comment]
> I'll like to work on this issue. Permit me to ask you any question if I get stuck or find anything strange.

You can work on any issue without asking, as long as there is no open PR linked to it (see #1092).

**Comment by benjaminmah** [comment]
I've created a [spreadsheet](https://docs.google.com/spreadsheets/d/1-29_OXtjfVwWsF03DyH2Q3dMNtYwNtB7VO7MLxNtXOI/edit?usp=sharing) compiling the False and True positives made by the SpamBug model, as well as the features that are sent to the model.

**Comment by marco-c** [comment]
@benjaminmah we should also look at feature importances, to try to understand what the model focused on for its wrong decision.

**Comment by marco-c** [comment]
@benjaminmah could you also add the product/component of the bug in the spreadsheet?

**Comment by benjaminmah** [comment]
> @benjaminmah we should also look at feature importances, to try to understand what the model focused on for its wrong decision.

Yes, I can definitely look into that! Being able to pinpoint which features are causing the false positives to occur the most would be helpful.


> @benjaminmah could you also add the product/component of the bug in the spreadsheet?

Yes, I'll add that ASAP.

**Comment by benjaminmah** [comment]
@marco-c The spreadsheet is updated with the product and component of each bug!

**Comment by marco-c** [comment]
Judging from the product, I think #2786 will help a lot here. We can re-evaluate this issue once that's done and deployed for a while.
